### ISSUE TITLE
High contrast theme implementation

### ISSUE THREAD

**Issue by andrew-ronaldson** [patternfly/patternfly#7448]
## Problem statement

Users with specific vision requirements find the interface difficult to read and navigate because the existing theme lacks sufficient color contrast. _"As a PatternFly user, I would like to have more accessible options for my vision requirements. (from the [patternfly-design epic](https://github.com/patternfly/patternfly-design/issues/1357))"_

## Risks

* Underestimation of Effort: The work required can be easily underestimated. A comprehensive audit of every single UI element and its various states (hover, focus, disabled, active, etc.) is a massive undertaking that could exceed initial timelines and budget.
* Scope Creep: The project could easily creep into refactoring the structure or styling of components
* Lack of Expertise: The resulting theme may fail to meet WCAG guidelines or could be implemented in a way that is not robust or scalable.
* This could introduce visual regressions in the default appearance of components

## Goal

Our users will have an accessible, high-contrast theme option that makes our product significantly easier to see, read, and use. This theme will be our first offering towards the larger goal of accessible themes that empower all users.

This epic focuses on the implementation of the high contrast color palette tokens and border styles as defined in https://github.com/patternfly/patternfly-design/issues/1376 and https://github.com/patternfly/patternfly-design/issues/1375

## Acceptance criteria

* New high contrast tokens from design are pulled into core. Core offers a mechanism to enable high contrast theme. All components/layouts/utilities reflect the appropriate, accessible high contrast presentation.
* Org has complete documentation for high contrast tokens. 
* Open question: do we include a toggle for displaying the site in high contrast?
* Communications are created for publicizing this feature.

## Relevant links

* Design epic - https://github.com/patternfly/patternfly-design/issues/1375
* Design figma - https://www.figma.com/design/c9C6sZWzBdbukivvXYiMkz/PF---High-Contrast-Theming?node-id=279-2121872&t=EwVEtNrTRDSbNDa8-1
* HC theme brainstorming - https://docs.google.com/document/d/14GFOdg0B_uq0DLq4beyaOm_x13cL4TevSQzmoFX20pk/edit?usp=sharing

## Level of effort
* 5 sprints, 2 devs

Open questions
* Should design weigh in on updating/creating the toggle switcher to add a high contrast toggle alongside dark theme switcher?

## Notes

High contrast design rules around color contrast and use of borders can be found in the [figma document under "rules"](https://www.figma.com/design/iT76DS020Ry1Wswfc0Hmes/branch/wKcOq7IrzfL1gEK2mocd6r/High-Contrast-Exploration?m=auto&node-id=10989-59477&t=de9CM89qsVdBP75x-1))

### General
- [x] #7624
- [x] https://github.com/patternfly/patternfly/issues/7737
- [x] https://github.com/patternfly/patternfly/issues/7765
- [x] https://github.com/patternfly/patternfly/issues/7779
- [x] #7496

### Border changes
- [x] #7592
- [x] #7611
- [x] #7612
- [x] #7594
- [x] #7589
- [x] #7591
- [x] #7596
- [x] #7613
- [x] #7614
- [x] #7597
- [x] #7615
- [x] #7616
- [x] #7617
- [x] #7618
- [x] #7619
- [x] #7600
- [x] #7620
- [x] #7601
- [x] #7621
- [x] #7590
- [x] #7602
- [x] #7603
- [x] #7604
- [x] #7605
- [x] #7622
- [x] #7623
- [x] #7608
- [x] #7609

#### Follow-ups/bugs
- [x] https://github.com/patternfly/patternfly/issues/7780
- [x] https://github.com/patternfly/patternfly/issues/7743

#### Org/website
- [x] https://github.com/patternfly/patternfly-org/issues/4738
- [x] https://github.com/patternfly/patternfly-org/issues/4759

**Comment by lboehling** [comment]
Hey @andrew-ronaldson here's a more thorough list of components needing new border styles:

- **About modal** 
- **Accordion** - regular variant: hover & expand states
- **Alerts** - overflow footer: default, hover/click states
- **Badge** - unread state - default
- **Banner** - bottom (or top) border - dependent on location
- **Button** - Link button, plain button, plain CTA: hover & clicked states
- **Cards** - secondary variant: default, hover, & clicked states.
- **Clipboard copy** - inline compact
- **Code block**
- **Code editor** - header
- **Calendar month** - panel + current day
- **Drawer** - overlay variants
- **Dual list selector** - hovered & selected items
- **Label** - overflow label variant & nonstatus filled label: default & hover states(exception to general border rule, let’s apply our nonstatus border color tokens here if possible in hc theme) 
- **Login page** - card
- **Menu toggle** - plain text and plain icon toggles (hover/clicked)
- **Menu** - menu items (hover/selected states), menu container, sticky footer
- **Modal** - container border
- **Navigation** - nav item hover & clicked. horizontal sub nav container.
- **Page** - main page border 
- **Panel** - raised/secondary variants
- **Popover** 
- **Progress bar** 
- **Simple list** - hover/clicked states
- **Table** - column headers hover/click states, draggable row active drag state
- **Tab**- hover states
- **Tree view** - hover/expanded states
- **Wizard** - step hover/clicked states, wizard container


I'm starting to spec these out in figma.

**Comment by mcoker** [comment]
Weekly update

* Dev work has begun 🥳
* Made updates to the POC - https://patternfly-react-pr-11911.surge.sh/patterns/high-contrast/react-demos/high-contrast/
  * Core PR style updates are in this branch - https://github.com/mcoker/patternfly/tree/hc-theme-poc-styles
  * React PR for the POC - https://github.com/patternfly/patternfly-react/pull/11911
* Created issues for remaining work from design ("Border changes" and "Focus changes" in this epic)
* Chatted with design about focus changes, most of which are still in the design phase. Will keep an eye on those as the design work progresses for anything that looks like it may impact our timeline.
* Still waiting for repos to be unfrozen to merge HC tokens in core - https://github.com/patternfly/patternfly/pull/7555/files

**Comment by mcoker** [comment]
Weekly update

Not much to report, we have been busy with the Q2 release and addressing issues found by testers. @srambach has kept the momentum going with a couple of PRs though!! 🥳
* https://github.com/patternfly/patternfly/pull/7656
* https://github.com/patternfly/patternfly/pull/7655

**Comment by mcoker** [comment]
Weekly update

Q2 release wrapped yesterday, so we're getting back on track with Hi-C.

* Created a [high contrast feature branch](https://github.com/patternfly/patternfly/tree/high-contrast-q3) for the time being.
* Have some open questions for design
  * When will we release given the focus style changes are slated for Q4. This may impact whether the work is continued on the feature branch or on the main branch. This involves considerations like releasing without announcing, releasing as "beta", etc.
  * Do we want to have status calls with design/dev and/or other stakeholders (Mary with Ansible)
*  Cleaned up the issue list, included correct figma links in issues, made sure issues are in the right buckets in GH (milestones, statuses, etc)
* A few more PRs are up
* Brainstorming some Hi-C border best practices so other devs can have something to reference if reviewing/contributing.

**Comment by mcoker** [comment]
Weekly update
* Added [HC developer guidance/best practices doc](https://docs.google.com/document/d/1jrsGm60BJHPBmuiXhMLDME82Iu0nAgCgKnizaxV8cNk/edit?usp=sharing). This is a live document and will be changing as we iterate on the solution.
* Closed the issues associated with the [q4 epic for focus style changes](https://github.com/patternfly/patternfly/issues/7658). We'll create those issues when they come up.
* Continuing figuring out the architecture as we go and see what issues arise:
  * Added support for [HC border width tokens](https://github.com/patternfly/patternfly/pull/7704)
  * Chatted with design about token naming and proposing we have tokens built in to default theme that are there to enable users (and us) to boost contrast, and communicate that they are not for the "PF high contrast" theme specifically.
* Lots of PRs in progress. Austin and Rebecca have contributed as well.
* Rebecca has added functionality to the HC POC that will be used for testing.

**Comment by mcoker** [comment]
Weekly update
* Updated [HC developer guidance/best practices doc](https://docs.google.com/document/d/1jrsGm60BJHPBmuiXhMLDME82Iu0nAgCgKnizaxV8cNk/edit?usp=sharing) with new recommendations based on addition of border-width tokens.
* Lots of issues being worked on and PRs up. Shout out to the whole team for contributing.
* [Updated the HC tokens](https://github.com/patternfly/patternfly/pull/7731) from design that fixed at least one bug we found in development.
* We had our kickoff meeting and talked through a number of things, a couple of updates from that meeting:
  * Release plan for this work in Q3 - planning on releasing what we have (token updates, HC borders) as alpha for the Q3 release, then the plan is for the full release in Q4 which will include focus styles, charts, chatbot, and a few other extensions.
  * Chatted through some options for charts style updates beyond just colors, which have some gotchas because charts are built 1) with a library (not just our code), 2) they're created with SVGs that have different style capabilities compared to regular html elements, and 3) in a couple of cases, we use apache eCharts, too, which might introduce some more gotchas.
* Opened issues for Q4
  * https://github.com/patternfly/react-catalog-view/issues/87 - Catalog view
  * https://github.com/patternfly/chatbot/issues/643 - Chatbot
  * https://github.com/patternfly/patternfly/issues/7724 - Charts
  * https://github.com/patternfly/patternfly-quickstarts/issues/409 - Quickstarts
  * https://github.com/patternfly/react-topology/issues/269 - Topology

**Comment by mcoker** [comment]
Weekly update
* Progress - 2 issues left that are unassigned, and only 2 of the assigned issues do not have an open PR. 11 open PRs.
* Met with design to work through new HC theme border tokens. @lboehling has the changes in figma, so we'll work this next sprint to implement them.
* New issues:
  * Add HC theme switcher (org/docs-framework) - https://github.com/patternfly/patternfly-org/issues/4738
  * Audit HC border use - https://github.com/patternfly/patternfly/issues/7737

**Comment by mcoker** [comment]
Weekly update

* No major updates, wrapping up dev work. A list of open and merged PRs - https://github.com/patternfly/patternfly/pulls?q=is%3Apr+base%3Ahigh-contrast-q3+

**Comment by mcoker** [comment]
Weekly update
* Wrapping up last issue - https://github.com/patternfly/patternfly/issues/7590
* HC toggle was added to the theme switcher in docs-framework
* Design collab to create token for the page's main content border

**Comment by mcoker** [comment]
Weekly update
* Component work is complete
* Theme switcher added to docs framework
* HC feature branch merged into main
* Added tokens for page content borders - good for default theme but makes HC theme switch easier
* Chatted about how to expose on org with release
  * Theme switcher will have beta label
  * Release notes will call out HC as beta
  * Talking with Lucia/Erin about any add'l content we might want on the website with the release

**Comment by mcoker** [comment]
Weekly update
* Design review done
* Bugs opened from design review, these are the remaining open ones
  * https://github.com/patternfly/patternfly/issues/7805
  * https://github.com/patternfly/patternfly/issues/7795
  * https://github.com/patternfly/patternfly/issues/7796 (sidebar panel issue)
* Moved the theme switcher to bottom/left in docs framework to fix a popper issue - https://github.com/patternfly/patternfly-org/pull/4775
* Made updates (still in progress) to the theme switcher menu for the website - https://github.com/patternfly/patternfly-org/pull/4766

**Comment by mcoker** [comment]
Weekly update
* Issues from design review all closed
  * https://github.com/patternfly/patternfly/issues/7792
  * https://github.com/patternfly/patternfly/issues/7793
  * https://github.com/patternfly/patternfly/issues/7795
  * https://github.com/patternfly/patternfly/issues/7796
  * https://github.com/patternfly/patternfly/issues/7797
  * https://github.com/patternfly/patternfly/issues/7798
  * https://github.com/patternfly/patternfly/issues/7804
  * https://github.com/patternfly/patternfly/issues/7805
* Made more updates to org theme switcher, still in progress - https://github.com/patternfly/patternfly-org/pull/4766
* Talked with team about Windows HCM/Firefox HC/`forced-colors: active` mode and identified some low-effort/high-impact bugs - https://github.com/patternfly/patternfly/issues/7830

**Comment by mcoker** [comment]
Weekly update
* Dev work is complete 🥳
* Did an [audit of all HC token changes](https://docs.google.com/document/d/1GqVv3_kG1X5u5MaIjnPk4BV8A-8Sp4iobYhM10BtJ9I/edit?usp=sharing) from the quarter
  * Found and resolved 4 bugs noted in https://github.com/patternfly/patternfly/issues/7864
* Resolved the WHCM/forced-colors issues noted in https://github.com/patternfly/patternfly/issues/7830

**Comment by mcoker** [comment]
All issues closed
